x-pack/plugins/security_solution/public/common/lib/keury/index.ts

```diff
@@ -80,20 +80,23 @@ export const convertToBuildEsQuery = ({
   indexPattern: IIndexPattern;
   queries: Query[];
   filters: Filter[];
-}) => {
+}): [string | undefined, Error | undefined] => {
```

Comment:
If the return value should only be one of the following:

- `[string, undefined]`
- `[undefined, Error]`

consider changing the return type from:

```
[string | undefined, Error | undefined]
```

to:

```
[string, undefined] | [undefined, Error]
```

because the latter form catches more invalid cases at compile time, and helps document intent. Note that even with this change, typescript can't catch every invalid combination within the `catch` block, because `error` actually is of type `any`, and TS [doesn't yet allow type annotations on catch clause variables](https://github.com/microsoft/TypeScript/issues/20024):

```ts
  } catch (error) {
    return ['foo', error]; // the compiler still won't catch this, because error is type `any`
  }
```